scripts/google_resumable_upload.py

```diff
@@ -0,0 +1,72 @@
+from google.auth.transport.requests import AuthorizedSession
+from google.resumable_media import requests, common
+import logging
+
+
+class GCSObjectStreamUpload(object):
```

Comment:
can you add some docstrings and explanation of what this class is? is this based on some Google-provided sample code?